anthos/deploy_admin_ws.sh

```diff
@@ -1,5 +1,37 @@
 #!/bin/bash
 
+# print a message to stderr, prefixed by HOSTNAME
+function note() {
+  echo 1>&2 "$HOSTNAME: $*"
+}
+
+# print the given command to stderr, run it, and exit verbosely if it fails.
+function xrun() {
+  vrun "$@" && return 0
+
+  local xstat=$?
+  note "Cmd $1 failed, exit $xstat"
+  exit "$xstat"
+}
+
+# print the given command to stderr, run it.  return the status
+function vrun() {
+  note "+ $@"
+  "$@"
+}
+
+# fetch the vcenter pem from the givem host:port, and print it to stdout.
+function fetch_pem() {
+  local addr=$1
+  xrun openssl s_client -showcerts -verify 5 -connect "$addr" < /dev/null \
+  | awk '/BEGIN/,/END/ {print}'
+  return "$((PIPESTATUS[0]))"
+}
+
+# ----- statt of mainline code
+
+HOSTNAME=$(hostname)
```

Comment:
HOSTNAME is a common environment.  While this will likely match $(hostname) already, I would rather not override HOSTNAME like this. LOG_PREFIX perhaps?